src/app/app-platform/ContentAppPlatform.h

```diff
@@ -81,11 +81,13 @@ class DLL_EXPORT ContentAppPlatform
     // add apps to the platform.
     // This will assign the app to an endpoint (if it is not already added) and make it accessible via Matter
     // returns the global endpoint for this app, or 0 if an error occurred
-    //
+    // If a desiredEndpointId is passed (cannot be less that Fixed endpoint count)
+    // framework will attempt to add the endpoint and assign the desiredEndpointId
+    // if desiredEndpointId is already taken endpoint is not added and 0 is returned.
     // dataVersionStorage.size() needs to be at least as big as the number of
     // server clusters in the EmberAfEndpointType passed in.
     EndpointId AddContentApp(ContentApp * app, EmberAfEndpointType * ep, const Span<DataVersion> & dataVersionStorage,
-                             const Span<const EmberAfDeviceType> & deviceTypeList);
+                             const Span<const EmberAfDeviceType> & deviceTypeList, EndpointId desiredEndpointId = 0);
```

Comment:
is 0 a correct default or should `kNoCurrentEndpointId`  be used?

Reply:
removed defaults.